src/util/dev/server.ts

```diff
@@ -329,11 +347,70 @@ export default class DevServer {
       }
     }
 
+    const apiFiles = await getApiFiles(this.cwd, this.output);
+
+    if (apiFiles.length > 0) {
+      if (!config.builds || config.builds.length === 0) {
+        config.builds = [];
+
+        if (pkg) {
+          const staticBuilder = await detectBuilder(pkg);
+
+          if (staticBuilder) {
+            config.builds.push(staticBuilder);
+          }
+        }
+
+        if (apiFiles && apiFiles.length > 0) {
+          const apiBuilds = await detectApiBuilders(apiFiles);
+
+          if (apiBuilds && apiBuilds.length > 0) {
+            config.builds.push(...apiBuilds);
+          }
+        }
+      }
+
+      const { defaultRoutes, error } = await detectApiRoutes(apiFiles);
+
+      if (error) {
+        this.output.error(error.message);
+      } else if (defaultRoutes && defaultRoutes.length > 0) {
```

Comment:
Ditto on [this comment](https://github.com/zeit/now-cli/pull/2498/files#r301260762) but towards `detectApiRoutes()`.

Reply:
Same as above, it can be `null` which is required by other consumers